File: src/main/java/com/github/unchama/util/Inventory.kt
package com.github.unchama.util

import arrow.core.Either
import com.github.unchama.menuinventory.InventorySize
import com.github.unchama.menuinventory.rows
import org.bukkit.Bukkit
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.InventoryHolder

/**
 * 主にチェスト用。
 */
inline val Inventory.row: Int
  get() = size / 9

fun createInventory(holder: InventoryHolder? = null,
                    size: InventorySize = 4.rows(),
                    title: String? = null): Inventory =
    when (size) {
      is Either.Left -> Bukkit.createInventory(holder, size.a.rows * 9, title)
      is Either.Right -> Bukkit.createInventory(holder, size.b, title)
    }

File: src/main/java/com/github/unchama/util/Inventory.scala
package com.github.unchama.util

import com.github.unchama.menuinventory.InventoryRowSize.InventorySize
import org.bukkit.Bukkit
import org.bukkit.inventory.{Inventory, InventoryHolder}
object InventoryUtil {
  import com.github.unchama.menuinventory.InventoryRowSize.IntToInventorySize
  /**
   * 主にチェスト用。
   */
  implicit class InventoryExtension(val inventory: Inventory) {
    def row: Int = inventory.getSize / 9
  }

  def createInventory(holder: InventoryHolder = null,
                      size: InventorySize = 4.rows,
                      title: String = null): Inventory =
    size match {
      case Left(a) => Bukkit.createInventory(holder, a.rows * 9, title)
      case Right(b) => Bukkit.createInventory(holder, b, title)
    }
}
